mig-agent/api/api.go

```diff
@@ -0,0 +1,148 @@
+package api
```

Comment:
I think it's fine in this case because it's scoped under `mig-agent/api`. There shouldn't be any confusion between `mig/api` and `mig/mig-agent/api` but if there is, one can always rename the import by writing
```go
import (
  apiInterface "github.com/mozilla/mig/mig-agent/api"
)

// ...

apiInterface.NewClient(...) // instead of api.NewClient
```